packages/create-subscription/README.md

```diff
@@ -0,0 +1,173 @@
+
+## Subscribing to event dispatchers
+
+Below is an example showing how `create-subscription` can be used to subscribe to event dispatchers such as DOM elements or Flux stores.
+
+```js
+import React from "react";
+import createComponent from "create-subscription";
+
+// Start with a simple component.
+// In this case, it's a functional component, but it could have been a class.
+function FollowerComponent({ followersCount }) {
+  return <div>You have {followersCount} followers!</div>;
+}
+
+// Create a wrapper component to manage the subscription.
+const EventHandlerSubscription = createComponent({
+  getValue: eventDispatcher => eventDispatcher.value,
+  subscribe: (eventDispatcher, callback) => {
+    const onChange = event => callback(eventDispatcher.value);
+    eventDispatcher.addEventListener("change", onChange);
+    return onChange;
+  },
+  unsubscribe: (eventDispatcher, subscription) => {
+    eventDispatcher.removeEventListener("change", subscription);
+  }
+});
+
+// Your component can now be used as shown below.
+// In this example, 'eventDispatcher' represents a generic event dispatcher.
+<EventHandlerSubscription source={eventDispatcher}>
+  {value => <FollowerComponent followersCount={value} />}
+</EventHandlerSubscription>
+```
+
+## Subscribing to observables
+
+Below are examples showing how `create-subscription` can be used to subscribe to certain types of observables (e.g. RxJS `BehaviorSubject` and `ReplaySubject`).
+
+**Note** that it is not possible to support all observable types (e.g. RxJS `Subject` or `Observable`) because some provide no way to read the "current" value after it has been emitted.
+
+### `BehaviorSubject`
+```js
+const BehaviorSubscription = createComponent({
+  getValue: behaviorSubject => behaviorSubject.getValue(),
+  subscribe: (behaviorSubject, callback) =>
+    behaviorSubject.subscribe(callback),
+  unsubscribe: (behaviorSubject, subscription) => behaviorSubject.unsubscribe()
+});
+```
+
+### `ReplaySubject`
+```js
+const ReplaySubscription = createComponent({
+  getValue: replaySubject => {
+    let currentValue;
+    // ReplaySubject does not have a sync data getter,
+    // So we need to temporarily subscribe to retrieve the most recent value.
+    replaySubject
+      .subscribe(value => {
+        currentValue = value;
+      })
+      .unsubscribe();
+    return currentValue;
+  },
+  subscribe: (replaySubject, callback) =>
+    replaySubject.subscribe(callback),
+  unsubscribe: (replaySubject, subscription) => replaySubject.unsubscribe()
+});
+```
+
+## Subscribing to a Promise
+
+Below is an example showing how `create-subscription` can be used with native Promises.
+
+**Note** that it an initial render value of `undefined` is unavoidable due to the fact that Promises provide no way to synchronously read their current value.
```

Comment:
> What do you think of showing an example that passes through the value returned by the promise?

I dig it.

> True; unless you're overly clever! :D

Unfortunately, that wouldn't prevent the _initial_ render with `undefined` because we don't "subscribe" until the commit phase. (Unless I'm misunderstanding something.)